sdk/core/azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -96,7 +96,96 @@ def on_request(self, request, **kwargs):
         request.http_request.headers.update(self.headers) # type: ignore
         additional_headers = request.context.options.pop('headers', {}) # type: ignore
         if additional_headers:
-            request.http_request.headers.update(additional_headers) # type: ignore
+            request.http_request.headers.update(additional_headers)  # type: ignore
+
+
+class TelemetryPolicy(SansIOHTTPPolicy):
+    """Telemetry policy for monitoring what SDK language, client library version,
+    and language/platform info a client is using to call into their service.
+
+    Clients can prepend additional information indicating the name and version of the client application.
+
+    :param str package_user_agent: The package information in the following format: <package_name>/<package_version>.
+        Example: "storage/4.0.0"
+
+    **Keyword arguments:**
+
+    *user_agent_overwrite (bool)* - Overwrites User-Agent when True. Defaults to False.
+
+    *user_agent_use_env (bool)* - Gets user-agent from environment. Defaults to True.
+    """
+
+    _USERAGENT = "User-Agent"
+    _ENV_ADDITIONAL_USER_AGENT = 'AZURE_HTTP_USER_AGENT'
+    _X_MS_AZSDK_Telemetry = "X-MS-AZSDK-Telemetry"
+
+    def __init__(self, package_user_agent=None, **kwargs):
+        # type: (Optional[str], Any) -> None
+        self.overwrite = kwargs.pop('user_agent_overwrite', False)
+        self.use_env = kwargs.pop('user_agent_use_env', True)
+        self.package_user_agent = package_user_agent
+
+        if package_user_agent is None:
+            self._user_agent = "azsdk-python-core/{} Python/{} ({})".format(
+                azcore_version,
+                platform.python_version(),
+                platform.platform(),
+            )
+        else:
+            self._user_agent = "azsdk-python-{} Python/{} ({})".format(
+                package_user_agent,
+                platform.python_version(),
+                platform.platform(),
+            )
```

Comment:
I believe the code inside conditional could be reduced by structuring way:

```
if package_user_agent is None:
    package_user_agent = "core/{}"format(azcore_version)

self._user_agent = "azsdk-python-{} Python/{} ({})".format(
    package_user_agent,
    platform.python_version(),
    platform.platform(),
)
```

At least for me, this makes it quicker/easier to see what exactly varies in the two cases.

Reply:
Agreed, that is much more clear. Changed my code to the above.